specs/latest/2.0/index.html

```diff
@@ -699,12 +694,23 @@ <h2 class="no-toc">Table of contents</h2>
   const GLenum MAX_CLIENT_WAIT_TIMEOUT_WEBGL                 = 0x9247;
 
   /* Buffer objects */
+  void bufferData(GLenum target, BufferDataSource? srcData, GLenum usage,
+                  optional unsigned long srcOffset = 0,
+                  optional unsigned long length = 0);
+  void bufferSubData(GLenum target, GLintptr dstByteOffset,
+                     BufferDataSource srcData,
+                     optional unsigned long srcOffset = 0,
+                     optional unsigned long length = 0);
+
```

Comment:
At this point I think it's reasonable to make these non-nullable types, assuming that doing so actually works in the browsers' IDL code generators.

There's already precedent for some WebGL 1.0 entry points taking non-nullable arguments. See the ArrayBufferView argument in compressedTexImage2D in https://www.khronos.org/registry/webgl/specs/latest/1.0/ for example.

Changing these arguments to non-nullable ones in WebGL 2.0 -- if that works from a technical perspective -- will reduce the amount of code that needs to be written for these entry points, since the DOM bindings will be responsible for raising an exception if null is passed.